src/pointer.cr

```diff
@@ -395,7 +395,7 @@ struct Pointer(T)
   # ptr.address # => 0
   # ```
   def self.null
-    new 0_u64
+    new LibC::SizeT.new(0)
```

Comment:
1. `size_t` is a length type, you meant `intptr_t` (pointer size);
2. you must hack the compiler for this to compile on 32bit platforms (`Pointer.new(UInt64)` is a Crystal intrinsic).